README.md

```diff
@@ -2,92 +2,133 @@
 
-The following mappings are stored by the portal network. Together with the accumulator, they replicate the functionality of canonical indices.
-	- Canonical block index: `block_number -> (block_hash)`
-		- Fetch the block header associated with `block_hash`, and validate the header against the accumulator to verify that `block_hash` is accurate for the given `block_number`.
-	- Canonical transaction index: `tx_hash -> (block_hash, tx_index)`
-		- Fetch the block header associated with `block_hash`, and verify the header against the accumulator.
-		- Then, fetch the block body associated with the header and verify that `tx_hash` matches the transaction found at the `tx_index` in the transaction trie.
+We refer to the data that backs these two lookups as "Canonical Indices" as they serve as a type of database index for fast retrieval of this information which would otherwise require prohibitively expensive scanning of all historical blocks.
 
+The responsibility for storing the records that make up these two indices should be evently distributed across the nodes in the network.  Nodes must be able to choose how many records from these indices they wish to store.  The records must be distributed across the network in a manner that allows nodes to determine the appropriate nodes to query for an individual record.
 
-### Transaction Sending: Cooperative Transaction Gossip
+Transaction information returned from this network will need to include a merkle proof against the `Header.transactions_trie` for validation purposes.
 
-The goal of the transaction gossip network is to make sure all new transactions are made available to the miners so that they can be included in a block.
+Block number information returned from this network will need to include a merkle proof against the header accumulator.
 
-Stateless clients should be able to declare what percentage of transactions they want to process out of the set of all unmined and valid transactions (called _mempool_) based on the amount of resources they have, and should only receive that many transactions from other nodes.
 
-Stateless transaction validation involves checking accounts' balances and nonces, so the network will need to facilitate transmission of account proofs alongside each transaction.
+### Transaction Gossip Network: Sending Transactions
+
+The Transaction Gossip Network facilitates broadcasting new transactions for inclusion in a future block.
+
+Nodes in this network must be able to limit how much of the transaction pool they wish to process and gossip.
+
+The goal of the transaction gossip network is to make sure nodes can broadcast transaction such that they are made available to miners for inclusion in a future block.
+
+Transactions which are part of this network's gossip must be able to be validated without access to the Ethereum state. This is accomplished by bundling a proof which includes the account balance and nonce for the transaction sender.
```

Comment:
This section might deserve a note mentioning this is required in order to prevent DOS attacks.